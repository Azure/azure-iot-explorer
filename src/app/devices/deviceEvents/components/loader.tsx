/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { Spinner, SpinnerSize, Stack, MessageBarType, MessageBar } from '@fluentui/react';
import { ResourceKeys } from '../../../../localization/resourceKeys';

export const Loader: React.FC<{monitoringData: boolean}> = ({monitoringData}) => {
    const { t } = useTranslation();

    return (
        <>
            {monitoringData &&
                <MessageBar
                    messageBarType={MessageBarType.info}
                >
                    <Stack horizontal={true} tokens={{ childrenGap: 10 }}>
                        <div>{t(ResourceKeys.deviceEvents.infiniteScroll.loading)}</div>
                        {<Spinner size={SpinnerSize.small} />}
                    </Stack>
                </MessageBar>
            }
        </>
    );
};
