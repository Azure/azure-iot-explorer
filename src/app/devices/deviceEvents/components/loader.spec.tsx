/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import 'jest';
import * as React from 'react';
import { shallow, mount } from 'enzyme';
import { Loader } from './loader';

describe('Loader', () => {
    it('matches snapshot when loading', () => {
        expect(shallow(<Loader monitoringData={true}/>)).toMatchSnapshot();
    });

    it('matches snapshot when not loading', () => {
        expect(shallow(<Loader monitoringData={false}/>)).toMatchSnapshot();
    });
});