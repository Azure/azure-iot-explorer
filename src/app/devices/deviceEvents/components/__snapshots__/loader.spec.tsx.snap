// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader matches snapshot when loading 1`] = `
<Fragment>
  <StyledMessageBar
    messageBarType={0}
  >
    <Stack
      horizontal={true}
      tokens={
        Object {
          "childrenGap": 10,
        }
      }
    >
      <div>
        deviceEvents.infiniteScroll.loading
      </div>
      <StyledSpinnerBase
        size={1}
      />
    </Stack>
  </StyledMessageBar>
</Fragment>
`;

exports[`Loader matches snapshot when not loading 1`] = `<Fragment />`;
